src/components/themed/SimpleTextInput.tsx

```diff
@@ -188,57 +189,76 @@ export const SimpleTextInput = React.forwardRef<SimpleTextInputRef, SimpleTextIn
-              inputAccessoryViewID={inputAccessoryViewID}
-              secureTextEntry={secureTextEntry}
-            />
-          </InnerContainer>
-
-          <TouchContainer hitSlop={theme.rem(0.75)} accessible onPress={handleClearPress} testID={`${testID}.clearIcon`}>
-            <SideContainer size={rightIconSize}>
-              <CloseIconAnimated color={iconColor} size={rightIconSize} />
-            </SideContainer>
-          </TouchContainer>
-        </InputContainerView>
+        <OuterInputContainerView>
+          <InputContainerView disableAnimation={disableAnimation} focusAnimation={focusAnimation} scale={scale} marginRemProps={marginRemProps}>
+            <SideContainer size={leftIconSize}>{Icon == null ? null : <Icon color={iconColor} size={leftIconSize} />}</SideContainer>
+            <TouchableOpacity hitSlop={theme.rem(0.75)} accessible onPress={handleDonePress} testID={`${testID}.doneButton`}>
+              <SideContainer size={backIconSize}>
+                <ChevronBackAnimated color={iconColor} size={backIconSize} />
+              </SideContainer>
+            </TouchableOpacity>
+
+            <InnerContainer>
+              <InputField
+                accessible
+                ref={inputRef}
+                keyboardType={keyboardType}
+                returnKeyType={returnKeyType}
+                accessibilityState={{ disabled }}
+                autoFocus={autoFocus}
+                disableAnimation={disableAnimation}
+                focusAnimation={focusAnimation}
+                placeholder={placeholder}
+                placeholderTextColor={placeholderTextColor}
+                selectionColor={theme.textInputTextColor}
+                testID={`${testID}.textInput`}
+                textAlignVertical="top"
+                value={value}
+                // Callbacks:
+                onBlur={handleBlur}
+                onChangeText={onChangeText}
+                onFocus={handleFocus}
+                onSubmitEditing={handleSubmitEditing}
+                maxLength={maxLength}
+                // Other Props:
+                autoCapitalize={autoCapitalize}
+                autoCorrect={autoCorrect}
+                blurOnSubmit={blurOnSubmit}
+                inputAccessoryViewID={inputAccessoryViewID}
+                secureTextEntry={secureTextEntry}
+              />
+            </InnerContainer>
+
+            <TouchContainer hitSlop={theme.rem(0.75)} accessible onPress={handleClearPress} testID={`${testID}.clearIcon`}>
+              <SideContainer size={rightIconSize}>
+                <CloseIconAnimated color={iconColor} size={rightIconSize} />
+              </SideContainer>
+            </TouchContainer>
+          </InputContainerView>
+          {Platform.OS === 'ios' && (
+            <CancelButton focusAnimation={focusAnimation}>
+              <TouchContainer hitSlop={theme.rem(0.75)} accessible onPress={handleDonePress} testID={`${testID}.cancelButton`}>
```

Comment:
Touch component within a touch component? That doesn't seem right. Move this to be a child of ContainerView.